File: 0005/0005.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        strLength = len(s)
        if (strLength == 1):
            return s
        start, end, maxSubStringLength = 0, 0, 1
        isPalindrome = [[False for i in range(strLength)] for i in range(strLength)]
        right = 1
        while(right < strLength):
            left = 0
            while(left < right):
                if (s[left] == s[right] and (right-left <= 2 or isPalindrome[left + 1][right - 1])):
                    isPalindrome[left][right] = True
                    if (right - left + 1 > maxSubStringLength):
                        maxSubStringLength = right - left + 1
                        start = left
                        end = right
                left += 1
            right += 1
        return s[start: end + 1]
A = Solution()
print(A.longestPalindrome("cbbd"))

        

File: 0005/0005.cs
public class Solution {
    public string LongestPalindrome(string s) {
        int StringLength = s.Length;
        if (StringLength == 1) {
            return s;
        }
        int start = 0;
        int maxSubStringLength = 1;
        bool[,] state = new bool[StringLength, StringLength];

        for (int right = 1; right < StringLength; right++) {
            for (int left = 0; left < right; left ++) {
                if (s[left] == s[right] && (right - left <= 2 || state[left, right])) {
                    state[left - 1, right + 1] = true;
                }
                if (right - left + 1 > maxSubStringLength) {
                    maxSubStringLength = right - left + 1;
                    start = left;
                }
            }
        }
        return s.Substring(start, maxSubStringLength);
    }
}

class A {
    public static void Main() {
        Solution sol = new Solution();
        Console.WriteLine(sol.LongestPalindrome("aaaa"));
    }
}